src/Haskell/Ide/Engine/Plugin/GhcMod.hs

```diff
@@ -290,6 +293,66 @@ cmp a b
 isSubRangeOf :: Range -> Range -> Bool
 isSubRangeOf (Range sa ea) (Range sb eb) = sb <= sa && eb >= ea
 
+
+
+data DocumentPosition =
+  DocPos {
+        docFile :: Uri
+      , docPos  :: Position
+      } deriving (Eq,Generic,Show)
+
+instance FromJSON DocumentPosition where
+  parseJSON = genericParseJSON $ defaultOptions { fieldLabelModifier = camelTo2 '_' . drop 3}
+instance ToJSON DocumentPosition where
+  toJSON = genericToJSON $ defaultOptions { fieldLabelModifier = camelTo2 '_' . drop 3}
+
+splitCaseCmd ::CommandFunc DocumentPosition WorkspaceEdit
+splitCaseCmd = CmdSync $ \(DocPos uri pos) -> do
+  splitCaseCmd' uri pos
+
+splitCaseCmd' :: Uri -> Position -> IdeGhcM (IdeResult WorkspaceEdit)
+splitCaseCmd' uri position =
+  pluginGetFile "splitCaseCmd: " uri $ \path -> do
+    origText <- GM.withMappedFile path $ liftIO . T.readFile
+    cachedMod <- getCachedModule path
+    case cachedMod of
+      ModuleCached checkedModule _ -> runGhcModCommand $ do
+        Just splitResult <- GM.splits' path (tcMod checkedModule) line column
+        return $ splitResultToWorkspaceEdit origText splitResult
+      ModuleFailed errText -> return $ IdeResultFail $ IdeError PluginError (T.append "hie-ghc-mod: " errText) Null
+      ModuleLoading -> return $ IdeResultOk mempty
```

Comment:
You should use `withCachedModule`, as that will automatically defer the action if the module is still loading.

Alternatively, you can defer it manually over here.

Reply:
I've looked at this but could not find a nice way to defer the action while keeping it in `IdeGhcM`. To defer an `IdeGhcM` action we'd need to use `IdeResponse'`, both of which do not work with `withCachedModule`. Furthermore, in the end all plugins seem to use `CommandFunc` which expects to return an `IdeResult`. I can't really find a sensible way to make these work together without creating a mess. Also note that other plugins like `typeCmd` don't defer either. Am I missing something?